tests/unit/test_sagemaker_steps.py

```diff
@@ -537,6 +597,63 @@ def test_training_step_creation_with_model(pca_estimator):
     }
 
 
+@patch('botocore.client.BaseClient._make_api_call', new=mock_boto_api_call)
+@patch.object(boto3.session.Session, 'region_name', 'us-east-1')
+def test_training_step_creation_with_model_with_env(pca_estimator_with_env):
+    training_step = TrainingStep('Training', estimator=pca_estimator_with_env, job_name='TrainingJob')
```

Comment:
This test is using a new model with a defined env, vpc_config and image_config 
vpc_config and image_config are not passed to ModelStep - this confirms that it is consistent with the current behaviour and will allow us to catch a breaking change.

Reply:
_(**not blocking:** but please follow this guidance for future contributions)_

You're following the existing conventions here, but let's try to structure the tests as documentation so they are easier to read in the future.
1. Name the test cases in a way that describes the functionality you are verifying. e.g. `test_training_step_get_expected_model_returns_model_with_environment`
2. Scope down the assertions to specific pieces if you're testing specific things. e.g. `assert training_step.get_expected_model(...).env == # the environment`. You can do have multiple and smaller asserts in a single test. 
3. If something is already covered in other test cases, you don't to repeat it every time.

Reply:
ACK
I'll take note to add these conventions in the CONTRIBUTING guide as well to ensure contributors have access to these